From 31b2578dcde4f4c1fa034f330a1062c8c2d69d4f Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Wed, 12 Aug 2015 11:08:44 +0100
Subject: [PATCH] Disable LPM Support in BOS descriptor.

This patch disables the LPM Support (and BESL Support)
inside the BOS descriptor, so that the device won't
support the L/U states.

Change-Id: I43b45f904cfbf12179777d282a9d05427ad6ee93
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42412
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
---
 drivers/usb/gadget/composite.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index f7d0a39..11f2302 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -528,7 +528,7 @@ static int bos_desc(struct usb_composite_dev *cdev)
 	usb_ext->bLength = USB_DT_USB_EXT_CAP_SIZE;
 	usb_ext->bDescriptorType = USB_DT_DEVICE_CAPABILITY;
 	usb_ext->bDevCapabilityType = USB_CAP_TYPE_EXT;
-	usb_ext->bmAttributes = cpu_to_le32(USB_LPM_SUPPORT | USB_BESL_SUPPORT);
+	usb_ext->bmAttributes = 0;
 
 	/*
 	 * The Superspeed USB Capability descriptor shall be implemented by all
-- 
1.9.1

